src/v/datalake/coordinator/coordinator.cc

```diff
@@ -311,6 +312,76 @@ coordinator::sync_ensure_table_exists(
     co_return std::nullopt;
 }
 
+ss::future<checked<std::nullopt_t, coordinator::errc>>
+coordinator::sync_ensure_dlq_table_exists(
+  model::topic topic, model::revision_id topic_revision) {
+    auto gate = maybe_gate();
+    if (gate.has_error()) {
+        co_return gate.error();
+    }
+
+    vlog(
+      datalake_log.debug,
+      "Sync ensure dlq table exists requested, topic: {} rev: {}",
+      topic,
+      topic_revision);
+
+    auto sync_res = co_await stm_->sync(10s);
+    if (sync_res.has_error()) {
+        co_return convert_stm_errc(sync_res.error());
+    }
+
+    // TODO: add mutex to protect against the thundering herd problem
+
+    topic_lifecycle_update update{
+      .topic = topic,
+      .revision = topic_revision,
+      .new_state = topic_state::lifecycle_state_t::live,
+    };
+    auto check_res = update.can_apply(stm_->state());
+    if (check_res.has_error()) {
+        vlog(
+          datalake_log.debug,
+          "Rejecting ensure_dlq_table_exist for {} rev {}: {}",
+          topic,
+          topic_revision,
+          check_res.error());
+        co_return errc::revision_mismatch;
+    }
+    if (check_res.value()) {
```

Comment:
For posterity, this is false if the topic lifecycle for this topic+revision is already live.